syntax = "proto3";
import "google/protobuf/timestamp.proto";
import "google/protobuf/descriptor.proto";

package tests;

extend google.protobuf.FieldOptions {
    bool is_optional = 50004;
}

// protoc --python_out=src -Isrc src/tests/sample.proto

message MessageOfTypes {

	double dubl = 1;
	float flot = 2;
	int32 i32 = 3;
	int64 i64 = 4;
	uint32 ui32 = 5;
	uint64 ui64 = 6;
	sint32 si32 = 7;
	sint64 si64 = 8;
	fixed32 f32 = 9;
	fixed64 f64 = 10;
	sfixed32 sf32 = 11;
	sfixed64 sf64 = 12;
	bool bol = 13;
	string strng = 14;
	bytes byts = 15;
	NestedType nested = 16;
	Enum enm = 17;
	repeated NestedType nestedRepeated = 18;
	repeated Enum enmRepeated = 19;
	map<string, float> simpleMap = 20;
	map<string, NestedType> nestedMap = 21;
	map<int32, int32> intMap = 22;

	message NestedType {
		string req = 1;
	}

	enum Enum {
		A = 0;
		B = 1;
		C = 2;
	}
}

enum Status {
    OK = 0;
    ERROR = 1;
}

message Obj {
    int32 id = 1 [(is_optional) = true];
    string item_id = 2;
    google.protobuf.Timestamp transacted_at = 3;
    Status status = 5;
}


message Objs {
    repeated Obj objects = 1;
}
